### DISCUSSION TITLE
Re-creating docker container removed world and backup tars contain an empty world

### DISCUSSION THREAD

**Discussion by mjuopperi** [mbround18/valheim-docker#239] [Help!]
Hi,

I re-created the container to persist a changed password. This ended up clearing the world. I did not empty the volumes my self, just got rid of the container.

Oh well, went to restore backup but they are all 1.4M and contain and empty world. I assume I'm fucked in regards to the world. Any idea what might cause the empty backups?

```
1.4M    backups/20210302-053001-backup.tar.gz
1.4M    backups/20210302-054501-backup.tar.gz
1.4M    backups/20210302-060001-backup.tar.gz
1.4M    backups/20210302-061501-backup.tar.gz
1.4M    backups/20210302-063001-backup.tar.gz
1.4M    backups/20210302-064501-backup.tar.gz
1.4M    backups/20210302-070001-backup.tar.gz
1.4M    backups/20210302-071501-backup.tar.gz
1.4M    backups/20210302-073001-backup.tar.gz
1.4M    backups/20210302-074501-backup.tar.gz
1.4M    backups/20210302-080001-backup.tar.gz
1.4M    backups/20210302-081501-backup.tar.gz
1.4M    backups/20210302-083001-backup.tar.gz
1.4M    backups/20210302-084501-backup.tar.gz
1.4M    backups/20210302-090001-backup.tar.gz
1.4M    backups/20210302-091501-backup.tar.gz
1.4M    backups/20210302-093001-backup.tar.gz
1.4M    backups/20210302-094501-backup.tar.gz
1.4M    backups/20210302-100001-backup.tar.gz
1.4M    backups/20210302-101501-backup.tar.gz
1.4M    backups/20210302-103001-backup.tar.gz
1.4M    backups/20210302-104501-backup.tar.gz
1.4M    backups/20210302-110001-backup.tar.gz
1.4M    backups/20210302-111501-backup.tar.gz
1.4M    backups/20210302-113001-backup.tar.gz
1.4M    backups/20210302-114501-backup.tar.gz
1.4M    backups/20210302-120001-backup.tar.gz
1.4M    backups/20210302-121501-backup.tar.gz
1.4M    backups/20210302-123001-backup.tar.gz
1.4M    backups/20210302-124501-backup.tar.gz
1.4M    backups/20210302-130001-backup.tar.gz
1.4M    backups/20210302-131501-backup.tar.gz
1.4M    backups/20210302-133001-backup.tar.gz
1.4M    backups/20210302-134501-backup.tar.gz
1.4M    backups/20210302-135024-pre-update-backup.tar.gz
```

**Comment by mjuopperi** [comment]
I've debugged this a little and it seems that the content of "saves" is not updated on my system.
I have selinux so I've mounted with the `:z` flag.

**Reply by mbround18** [reply]
What does the z flag do?

**Reply by mjuopperi** [reply]
It tells docker to set correct labels on the directory so that it can be accessed. It's a selinux thing. Works fine for my other containers and I don't see any permission issues on this one either, so I think that is irrelevant.

I've tested this more today and it seems like that save directory gets updated when I stop the server with odin but not otherwise. Any idea what might be causing this?

**Reply by mbround18** [reply]
What do you mean not otherwise? 0.o if you stop the container normally it should call odin to stop the server and save the world. However if you kill the container to stop immediately it will not run odin commands to save the world.

**Reply by mjuopperi** [reply]
I mean it does not save it periodically, so if the container crashes, the world is gone. I've been watching the directory with `inotifywait` and there are no writes unless I stop or start the server with odin (or with docker I guess).

**Reply by CosmicHorrorDev** [reply]
I'll try to verify this behavior, but that's entirely possibly.

Sadly we don't have a way to force the server to save the worlds current state since there aren't really any administration tools yet or else we would force a save and disable autosaving before we create the backup.

**Reply by mbround18** [reply]
Valheim should be saving the server periodically and on world creation. Could you try without the z flag?
